### DISCUSSION TITLE
Metadata generation

### DISCUSSION THREAD

**Discussion by marcinsiewiera** [ScalefreeCOM/turbovault4dbt#19] [Ideas]
Hi All, 

Have you thought about different approach, e.g. starting from annotated source data and generating respective DV objects?

E.g. Simplified input could look like in table below 
![obraz](https://github.com/ScalefreeCOM/turbovault4dbt/assets/132299743/001c8010-cd76-418a-a5b8-2f1601ec5839)
based on the annotated source you could generate metadata for all the DV objects.

Regards,
Marcin

**Comment by hle-scalefree** [comment]
Hello Marcin,
thanks for Your interesting comment on this!
The current template of the metadata sheet that is used for TurboVault4dbt could also be seen as a "compact" version of the annotated source data. Therefore, we kept the TurboVault4dbt template as is so that only necessary information is displayed.
However, I have seen in various projects where they used simple Excel-Functions / VBA scripts to convert the annotated source into the TurboVault4dbt sheet.

Best Regards,
Hoa

**Comment by marcinsiewiera** [comment]
Can you share a sample xls/vba you mentioned? I'm working on something similar and definitely would be able to contribute to a extension of the tool. This functionality would be a huge value add based on my experience.

**Reply by hle-scalefree** [reply]
Unfortunately, I do not have direct access to the source code thus I cannot share it. Sorry for that!
